### PR TITLE
Automate fastlane version bumps

### PR DESCRIPTION
**Pull request by KrauseFx**
- Show changelog
- Ask for type of version bump
- Increment the version number
- Commit the changes
- Push to GitHub
- Submit PR
- Add git label
- Post a message to slack

Sample PR: https://github.com/fastlane/fastlane/pull/7288

screenshot 2016-12-02 14 54 57

### CODE REVIEW

File: fastlane/fastlane/Fastfile [hunk 1]
```diff
@@ -6,28 +6,65 @@ end
...
 
-  if bump_type
-    version_bump_podspec(path: path, bump_type: bump_type)
-    git_commit(path: path,
-               message: "Version Bump by fastlane")
-    push_to_git_remote
-  else
-    puts "Doing nothing..."
+  github_api_token = ENV["FL_GITHUB_RELEASE_API_TOKEN"]
+  UI.user_error!("Please provide a GitHub API token using `FL_GITHUB_RELEASE_API_TOKEN`") if github_api_token.to_s.length == 0
```
Comment (milch) [inline fastlane/fastlane/Fastfile]: Commits and pushes are already happening before this, so the check should be at the beginning of the lane

File: fastlane/fastlane/Fastfile [hunk 2]
```diff
@@ -6,28 +6,65 @@ end
...
+  # Add a label to the pull request
+  pr_id = pr_url.match(%r{pull\/(\d+)})[1]
+
+  Actions::SetGithubReleaseAction.call_releases_endpoint(
+    "post",
+    "https://api.github.com",
+    slug,
+    "/issues/#{pr_id}/labels",
+    github_api_token,
+    [tool_name].to_json
```
Comment (milch) [inline fastlane/fastlane/Fastfile]: Maybe `"[#{tool_name}]"` instead? Not sure if the `.to_json` provides any additional value here

**Review by milch — DISMISSED** [review]
♻️

**Comment by mfurtak** [comment]
This is sweet! 🍬 

I like seeing the "Changes since release x.x.x" above the list of commits. Can we work that into the comment? 😄 🙏

File: fastlane/fastlane/Fastfile (hunk 2 above)
Comment (KrauseFx) [inline fastlane/fastlane/Fastfile]: The API call requires `.to_json` 👍

File: fastlane/fastlane/Fastfile (hunk 1 above)
Comment (KrauseFx) [inline fastlane/fastlane/Fastfile]: I'll update the PR, no worries

**Comment by KrauseFx** [comment]
Addressed feedback 👍

**Review by mfurtak — APPROVED** [review]
🤖 ❤️

**Pull request merged**